whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisClientCertificateAuthenticationTestIntegration.java

```diff
@@ -1,34 +1,41 @@
 package net.ripe.db.whois.api.rest;
 
-import net.ripe.db.whois.api.AbstractIntegrationTest;
+import jakarta.ws.rs.NotAuthorizedException;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.MediaType;
 import net.ripe.db.whois.api.RestTest;
+import net.ripe.db.whois.api.SecureRestTest;
+import net.ripe.db.whois.api.httpserver.AbstractClientCertificateIntegrationTest;
+import net.ripe.db.whois.api.httpserver.CertificatePrivateKeyPair;
+import net.ripe.db.whois.api.httpserver.WhoisKeystore;
+import net.ripe.db.whois.api.rest.domain.WhoisObject;
 import net.ripe.db.whois.api.rest.domain.WhoisResources;
 import net.ripe.db.whois.api.rest.mapper.FormattedClientAttributeMapper;
+import net.ripe.db.whois.api.rest.mapper.FormattedServerAttributeMapper;
 import net.ripe.db.whois.api.rest.mapper.WhoisObjectMapper;
-import net.ripe.db.whois.common.TestDateTimeProvider;
+import net.ripe.db.whois.common.domain.CIString;
 import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
 import net.ripe.db.whois.common.rpsl.RpslAttribute;
 import net.ripe.db.whois.common.rpsl.RpslObject;
 import net.ripe.db.whois.common.rpsl.RpslObjectBuilder;
-import net.ripe.db.whois.update.keycert.X509CertificateUtil;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 
-import jakarta.ws.rs.client.Entity;
-import jakarta.ws.rs.core.MediaType;
-import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLContext;
+import java.nio.file.Files;
+import java.nio.file.Path;
 
-import static net.ripe.db.whois.common.domain.CIString.ciString;
-import static net.ripe.db.whois.common.rpsl.ObjectType.ROUTE6;
-import static net.ripe.db.whois.update.keycert.X509CertificateUtil.getCertificateAsString;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.fail;
 
 @Tag("IntegrationTest")
-public class WhoisClientCertificateAuthenticationTestIntegration extends AbstractIntegrationTest {
+public class WhoisClientCertificateAuthenticationTestIntegration extends AbstractClientCertificateIntegrationTest {
```

Comment:
Rename by ClientCertificateAuthenticationTestIntegration?

Reply:
Good idea, but I renamed it to `WhoisRestServiceClientCertificateTestIntegration` to be consistent with the other REST test classes (it contains REST API tests).